qemu/tests/nvdimm.py

```diff
@@ -107,50 +108,68 @@ def run(test, params, env):
-            raise error.TestError("Check nv_backend in host failed!")
+            test.fail("Check nv_backend in host failed!")
         params["start_vm"] = "yes"
         vm_name = params['main_vm']
         env_process.preprocess_vm(test, params, env, vm_name)
 
-    nvdimm_test = NvdimmTest(params, env)
+    nvdimm_test = NvdimmTest(test, params, env)
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
     try:
-        error.context("Login to the guest", logging.info)
+        error_context.context("Login to the guest", logging.info)
         login_timeout = int(params.get("login_timeout", 360))
         nvdimm_test.session = vm.wait_for_login(timeout=login_timeout)
-        error.context("Verify nvdimm in monitor and guest", logging.info)
+        error_context.context("Verify nvdimm in monitor and guest", logging.info)
         nvdimm_test.verify_nvdimm(vm)
-        error.context("Format and mount nvdimm in guest", logging.info)
+        error_context.context("Format and mount nvdimm in guest", logging.info)
         nvdimm_test.mount_nvdimm()
         nv_file = params.get("nv_file", "/mnt/nv")
-        error.context("Create a file in nvdimm mount dir in guest, and get "
-                      "original md5 of the file", logging.info)
+        error_context.context("Create a file in nvdimm mount dir in guest, and get "
+                              "original md5 of the file", logging.info)
         dd_cmd = "dd if=/dev/urandom of=%s bs=1K count=200" % nv_file
         nvdimm_test.run_guest_cmd(dd_cmd)
         orig_md5 = nvdimm_test.md5_hash(nv_file)
         nvdimm_test.umount_nvdimm()
         nvdimm_test.session = vm.reboot()
-        error.context("Verify nvdimm after reboot", logging.info)
+        error_context.context("Verify nvdimm after reboot", logging.info)
         nvdimm_test.verify_nvdimm(vm)
         nvdimm_test.mount_nvdimm(format_device="no")
         new_md5 = nvdimm_test.md5_hash(nv_file)
-        error.context("Compare current md5 to original md5", logging.info)
+        error_context.context("Compare current md5 to original md5", logging.info)
         if new_md5 != orig_md5:
-            raise error.TestFail("'%s' changed. The original md5 is '%s',"
-                                 " current md5 is '%s'"
-                                 % (nv_file, orig_md5, new_md5))
+            test.fail("'%s' changed. The original md5 is '%s', current md5 is '%s'"
+                      % (nv_file, orig_md5, new_md5))
         nvdimm_test.umount_nvdimm()
-        error.context("Check if error and calltrace in guest", logging.info)
+        error_context.context("Check if error and calltrace in guest", logging.info)
         vm.verify_kernel_crash()
 
     finally:
+        if nvdimm_dax:
+            try:
+                process.system(params["del_dax_cmd"], timeout=240, shell=True)
+            except process.CmdError:
+                test.fail("Host dax configuration cannot be deleted!")
```

Comment:
IMHO, this operation, as a cleanup step, should be executed after VM process terminated, and the result of this operation should not affect the test result.

@huangyum what's your opinion?

Reply:
Yes, you are right.  The cleanup step on host should not affect the test result.

Reply:
Modified, Thanks! @luckyh